### PR TITLE
fix(eks): private endpoint access doesn't work with `Vpc.fromLookup`

### LINKED ISSUE #9542: [aws-eks] private endpoint doesn't work with looked up vpc
It's impossible to configure a private access endpoint when using a looked up vpc (i.e `Vpc.fromLookup`).

### Reproduction Steps

1. Create a VPC using the following guide: https://docs.aws.amazon.com/batch/latest/userguide/create-public-private-vpc.html

2. Deploy the following:

```typescript
const vpc = ec2.Vpc.fromLookup(this, 'Vpc', {
  vpcId: '<VPC_ID>'
})

const cluster = new eks.Cluster(this, 'Cluster', {
  vpc: vpc,
  version: eks.KubernetesVersion.V1_16,
  defaultCapacity: 2,
  endpointAccess: eks.EndpointAccess.PRIVATE
});

cluster.addResource('config-map', {
  kind: 'ConfigMap',
  apiVersion: 'v1',
  data: {
    hello: 'world',
  },
  metadata: {
    name: 'config-map',
  },
});
```

### What did you expect to happen?

Expected for the cluster to created and for the config map to be correctly applied

### What actually happened?

Screen Shot 2020-08-08 at 9 57 46 PM

### Environment

  - **CLI Version      :1.57.0**
  - **Framework Version:1.57.0**
  - **Node.js Version:** 
  - **OS               :**
  - **Language (Version):** 

### Other

Following is an excerpt from CloudFormation resource for the kubectl handler, we can see that the subnet id's are indeed empty:

```json
    "Handler886CB40B": {
      "Type": "AWS::Lambda::Function",
      "Properties": {
        ....
        "VpcConfig": {
          "SecurityGroupIds": [
            {
              "Ref": "referencetoekslookedupvpcPlaygroundClusterKubectlProviderSecurityGroupBFF8A75BGroupId"
            }
          ],
          "SubnetIds": []
        }
        ...
      },
    },

```

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by iliapolo**
Fixes #9542 
Related #5383 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by JasperW01** [comment]
put my 2 cents in the PR as well. 

Hi @iliapolo , thx a lot for raising this. really appreciate.

Just my 2 cents on the subnet selections for the kubectl handler Lambda given the enterprise context:

i. In most cases, we will configure vpcSubnets when calling eks.Cluster or FargateCluster to create eks clusters, so we can make sure the ENIs of the EKS control plane will be put into the subnets as required. But regardless the vpcSubnets is set or not, I guess by default the kubectl handler Lambda can just be associated with the subnets associated with the EKS Control plane.

ii. However, that might not be enough. This kubectl handler Lambda fundamentally is applying kubectl/helm, so arguably it's mostly doing the tasks of the data plane. So it would be great if we have a separate subnet selection option in eks.Cluster & eks.FargateClusters specifically for the kubectl handler Lambda, sth like kubectlSubnets along with kubectlEnvironment.

Having said that, I fully appreciate that you need to address this issue by considering the whole thing holistically. So please feel free to ignore this suggestion if it's contradicting to other factors you might need to consider.

Thx a lot in advance. Jasper

**Comment by iliapolo** [comment]
Hi @JasperW01 

My concern with providing a `kubectlSubnets` property is that it's somewhat error prone. A user might configure it with subnets that don't overlap with the control plane EKS subnets, in which case the kubectl handler won't be able to access the cluster. (haven't validated this yet).

As I understand, you are talking about an option to specify just a subset of private subnets from the cluster subnets, right? Do you have a specific use case where using all private subnets of the cluster would be somehow insufficient or harmful? 

Thanks

**Comment by JasperW01** [comment]
Hi @iliapolo , second thought after reading your feedback - to make it simple I think let's just treat the kubectl handler as part of the control plane, so we can just align its subnet association with whatever subnets selected for the control plane. 

Please allow me to recap our use case so we can be a bit clear, 

i. we have several artificial security zones, each security zone has one subnet per AZ. So we together we have many subnets, and as a general rule, each workload needs to be positioned to the 3 subnets corresponding to the security zone the workload is classified with. EKS cluster (both control plane and data plane) is no difference. 

ii. So when we create EKS cluster control plane in cdk code, we need to specify vpcSubnets in eks.Cluster/FargateCluster to make sure the two ENIs of the control plane are positioned within the 3 subnets corresponding to the proper security zone. 

iii. Then in cdk code we create EKS worker node groups or Fargate profiles separately and specify the subnetSelection to position the worker node groups or Fargate profiles to the subnets corresponding to their proper security zones respectively. 

iv. BTW, to make the above approach work, beforehand we already tagged all the required subnets with kubernetes.io/cluster/<cluster-name> = 'shared" & kubernetes.io/role/elb (internal-elb) = '1'. As a side point, this is sth we hope we don't need to do - but it's a problem for another day. 

So now let's put the kubectl handler lambda into this picture. To keep it simple, I think we can just treat it as part of the EKS control plane, i.e. associate such Lambda with the subnets selected for the EKS control plane.

**Comment by JasperW01** [comment]
Hi @iliapolo BTW, I double checked the code changes in this PR and I believe it will work in our context.

File: packages/@aws-cdk/aws-eks/lib/cluster.ts [hunk 1]
```diff
@@ -1049,7 +1056,26 @@ export class Cluster extends Resource implements ICluster {
...
+          privateSubnets.push(subnet);
+          continue;
+        }
+
+        if (this.vpc.publicSubnets.includes(subnet)) {
+          // definitely public, skip it.
+          continue;
+        }
+
+        // not public and not private - what is it then? this means its a subnet instance that was explicitly passed
```
Comment (eladb) [inline packages/@aws-cdk/aws-eks/lib/cluster.ts:1073]:
```suggestion
        // not public and nor private - what is it then? this means its a subnet instance that was explicitly passed
```

File: packages/@aws-cdk/aws-eks/lib/cluster.ts [hunk 2]
```diff
@@ -1049,7 +1056,26 @@ export class Cluster extends Resource implements ICluster {
...
+        // in the subnet selection. since ISubnet doesn't contain information on type, we have to assume its private and let it
+        // fail at deploy time :\ (its better than filtering it out and preventing a possibly succefull deployment)
```
Comment (eladb) [inline packages/@aws-cdk/aws-eks/lib/cluster.ts:1075]:
```suggestion
        // fail at deploy time :\ (its better than filtering it out and preventing a possibly successful deployment)
```

**Pull request merged**